File: SVN_REPO/assign2/erlang/test/fibonacci_test.erl
-module(fibonacci_test).
-include_lib("eunit/include/eunit.hrl").
-compile([export_all]).

run_fibonacci_tests(FibonacciFunction) ->
    [
        ?_assertEqual([1], FibonacciFunction(0)),
        ?_assertEqual([1, 1], FibonacciFunction(1)),
        ?_assertEqual([1, 1, 2], FibonacciFunction(2)),
        ?_assertEqual([1, 1, 2, 3, 5, 8], FibonacciFunction(5)),
        ?_assertEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89], FibonacciFunction(10))
    ].

fibonacci_test() ->
    run_fibonacci_tests(fun fibonacci:fibonacci_iterative/1).

fibonacci_recursive_test() ->
    run_fibonacci_tests(fun fibonacci:fibonacci_recursive/1).

fibonacci_recursive_iterative_test() ->
    run_fibonacci_tests(fun fibonacci:fibonacci_recursive_iterative/1).

File: SVN_REPO/assign2/scala/src/test/scala/fibonacci/FibonacciTest.scala
package fibonacci
import org.scalatest.funsuite.AnyFunSuite

class FibonacciTest extends AnyFunSuite {

  def testFibonacciFunction(fibonacciFunction: Int => Seq[Int], functionName: String): Unit = {
    test(s"Testing $functionName") {
      val testCases = Map(
        0 -> Seq(1),
        1 -> Seq(1, 1),
        2 -> Seq(1, 1, 2),
        5 -> Seq(1, 1, 2, 3, 5, 8),
        10 -> Seq(1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89)
      )

      testCases.foreach { case (position, expected) =>
        val result = fibonacciFunction(position)
        assert(result == expected, s"Expected $expected for position $position")
      }
    }
  }

  testFibonacciFunction(fibonacci_Iterative, "Iterative")
  testFibonacciFunction(fibonacci_Recursive, "Recursive")
  testFibonacciFunction(fibonacci_Recursive_Iterative(_), "RecursiveIterative")
}
